libs/src/sanityChecks/rolloverNodes.js

```diff
@@ -0,0 +1,55 @@
+const Utils = require('../utils')
+
+const TEN_SECONDS = 10000
+const MAX_TRIES = 3
+
+/** Check if the user's primary creator node is healthy */
+const checkPrimaryHealthy = async (libs, primary, tries) => {
+  const healthy = await Utils.isHealthy(primary)
+  if (healthy) return healthy
+  else {
+    if (tries === 0) {
+      return false
+    }
+    await Utils.wait(TEN_SECONDS)
+    return checkPrimaryHealthy(libs, primary, tries - 1)
+  }
+}
+
+/** Gets new endpoints from a user's secondaries */
+const getNewEndpoints = async (libs, secondaries) => {
+  for (const secondary of secondaries) {
+    const { status, userBlockNumber } = await libs.creatorNode.getSyncStatus(secondary)
```

Comment:
where is status.blocknumber defined? the sync_status endpoint returns `return successResponse({ walletPublicKey, latestBlockNumber })` - is it in the middleware somewhere? also, how does this comparison represent the latest known blocknumber across nodes?

Reply:
good catch, i think it should be `status.latestBlockNumber`. i'll fix.
this code is outdated, check out ray's new commits. `cnode.latestBlockNumber` is compared against `max(discprov.userBlockNumber, discprov.trackBlockNumber)`